test/db.js

```diff
@@ -217,34 +217,30 @@ test('Berlin Jungfernheide to ATZE Musiktheater', co(function* (t) {
 	t.end()
 }))
 
-test('journeys: via works – changing trains', co(function* (t) {
-	// Going from Westhafen to Wedding via Württembergalle without changing
-	// is currently impossible. We check if the routing engine lets us change
-	// at Württembergallee if we set it as via.
+test('journeys: via works – with detour', co(function* (t) {
+	// Going from Westhafen to Wedding via Württembergalle without detour
+	// is currently impossible
 	const westhafen = '008089116'
 	const wedding = '008089131'
 	const württembergallee = '731084'
 	const [journey] = yield client.journeys(westhafen, wedding, {
 		via: württembergallee,
 		results: 1,
-		when
+		when,
+		passedStations: true
 	})
 
 	t.ok(journey)
-	const i1 = journey.legs.findIndex(l => l.destination.id === württembergallee)
-	t.ok(i1 >= 0, 'no leg with Württembergallee as destination')
 
-	const i2 = journey.legs.findIndex(l => l.origin.id === württembergallee)
-	t.ok(i2 >= 0, 'no leg with Württembergallee as origin')
-	t.ok(i2 >= i1)
+	const l = journey.legs.some(l => l.passed && l.passed.some(p => p.station.id === württembergallee))
+	t.ok(l, 'no stopover at Württembergalle')
```

Comment:
"stopover" is ambiguous, because it can meant both "changing vehicles" and "stopping at, but staying in the vehicle". I think "Württembergallee ist not being passed" is also not the best phrasing, but matches the rest of the code base.